json/DataTypes/Decoding.roc

```diff
@@ -0,0 +1,167 @@
+module [string, number, bool, list, field, map, map2, map3, andThen, or, JsonDecoder, DecodingErrors]
+import JsonData exposing [Json]
+
+DecodingErrors : [
+    FieldNotFound Str,
+    ExpectedJsonObject Str,
+    ExpectedJsonArray Str,
+    WrongJsonType Str,
+    KeyNotFound,
+    DecodingFailed
+]
+    
+JsonDecoder t : Json -> Result t DecodingErrors
+
+string : JsonDecoder Str
+string = \json ->
+    when json is
+        String str -> Ok str
+        _ -> Err (WrongJsonType "Expected a String when decoding, found $(printJsonType json)")
```

Comment:
This looks great, and I wanted to note that calling a function to convert to `Str` inside the `$(...)` is definitely idiomatic! 💯 

Only thing I'd tweak here is to rename `printJsonType` to something like `typeToStr` just because I think of `print` meaning like "print to terminal" or something like that.

Reply:
Yep fair point!